nixos/tests/zfs.nix

```diff
@@ -1,5 +1,5 @@
 { system ? builtins.currentSystem,
-  config ? {},
+  config ? { boot.kernelPackages = pkgs.linuxPackages_latest; },
   pkgs ? import ../.. { inherit system config; }
```

Comment:
I think this should be a separate argument. The `config` here doesn't have anything to do with a NixOS configuration. Something like

```nix
{ system ? builtins.currentSystem
, config ? {}
, pkgs ? import ../.. { inherit system config; }
, kernelPackages ? pkgs.linuxPackages_latest }: null
```

Reply:
The purpose of doing this was precisely to let users (e.g. me) pass a NixOS configuration in. You can do that with a fair number of other tests, and it's basically the main reason to run tests locally.

Ideally I'd like to make it work for all of them, with the same interface, letting me add a --run-tests option to nixos-rebuild.

Reply:
I don't see any test using the nixpkgs config as a NixOS config? They all pass it to `import <nixpkgs> { inherit config; }`, where a NixOS config makes no sense at all.

If you want it to support passing in arbitrary NixOS configs you need to do something like this:

```nix
{ nixosConfiguration ? {}
, ...
}:
let
  kernelPackages = (import ../../nixos {
    configuration = nixosConfiguration;
  }).config.boot.kernelPackages;
in ...
```